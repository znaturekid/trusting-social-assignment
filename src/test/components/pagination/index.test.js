import React from 'react'
import Enzyme, {
    mount,
} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import PaginationComponent from '../../../components/pagination'
import 'jest-styled-components';

Enzyme.configure({
    adapter: new Adapter()
})

function setup() {
    const props = {
        nextPage: jest.fn(),
        page: 1,
        loading: false
    }

    const component = <PaginationComponent { ...props} />

    return {
        props,
        component
    }
}


function setupWithProps(customProps) {
    const props = {
        nextPage: jest.fn(),
        page: 1,
        loading: false,
        ...customProps
    }

    const component = <PaginationComponent { ...props} />

    return {
        props,
        component
    }
}


describe('components', () => {
    describe('pagination =>', () => {
        it('should render self and subcomponents', () => {
            const { component } = setup()
            expect(mount(component).text()).toBe('Load More');
        })

        it('should call search function after change event fire', () => {
            const { component, props } = setup()
            const button = mount(component);
            button.simulate('click');
            expect(props.nextPage).toBeCalled();
            expect(button.state().autoScroll).toBe(true)
        })

        it('should render button with loading text', () => {
            const { component } = setupWithProps({loading:true})
            expect(mount(component).text()).toBe('Loading...');
        })

        it('should render button with loading text', () => {
            const { component } = setupWithProps({page:200})
            expect(mount(component).text()).toBe('No More');
        })
    })
})