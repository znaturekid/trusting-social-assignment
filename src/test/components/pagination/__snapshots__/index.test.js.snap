// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Search= should render self and subcomponents 1`] = `
<styled.div
  data-loading={false}
>
  <styled.input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Enter text to search"
    type="text"
  />
</styled.div>
`;

exports[`components pagination= should render self and subcomponents 1`] = `
<styled.button
  disabled={false}
  onClick={[Function]}
>
  Load More
</styled.button>
`;

exports[`components pagination=> should render self and subcomponents 1`] = `
<styled.button
  disabled={false}
  onClick={[Function]}
>
  Load More
</styled.button>
`;
